projecttacoma/bulk-export-app
# Create Export Selection Page

# Summary
This PR creates the basic landing page for the bulk data app as well as the export selection page.

## New behavior
There is now a landing page on localhost:[port] that describes the purpose of the app. There is a button that can be used to navigate to the next page, the /query-selector page. Here, you can choose which type of query you want to start to build, either a patient-level, group-level or system-level. The patient and group ones have a searchable dropdown to select and id that you want to query. The button underneath is disabled until you select one. When you click one of these buttons, you will navigate to the query-builder page, which currently just displays the data in the query string of the url.

## Code changes
`/src/app/page.tsx` is the landing page
`/src/app/query-selector/page.tsx` is the query-selector page
`/src/app/query-builder/page.tsx` is the query-builder page which is mostly blank at the moment
`/src/components` is a folder which contains all the components for the query-selector page.
- `/src/components/export-type.tsx` is used for each of the three boxes on the query-selector page.
-  `/src/components/build-query-button.tsx` is the button at the bottom of each of the export-type components on the page
- `/src/components/dropdown.tsx` is the dropdown for either the group or patient export based on what is passed in as a name.
- `/src/components/group-dropdown.tsx` is the group specific dropdown which populates the group id data from the bulk-data-server.
- `/src/components/patient-dropdown.tsx` is the patient specific dropdown which for now is filled with test data that I made up. It will also be filled with the patient ids from the server once there is an endpoint that can get that data.

`/src/components/componentStyles` is a folder that contains css module files for each component to style them.
`/src/util` contains utility functions for the app. Currently just has one file, a string utility file.

# Testing guidance
- Start the bulk data server instructions [here](https://github.com/projecttacoma/bulk-export-server)
- `npm run check`
- `npm run dev`
- Go to http://localhost:3000/
- Navigate to query selector page
- Try selecting an id from the dropdown and click the button. Verify that the data on the next page is the selected data.
- Try selecting an id from both dropdowns and clicking the button on one of them. Make sure that the type of the export is correct.
- Verify that the button cannot be pressed if there is not a element selected, and that the tooltip appears.
